activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -110,5 +110,24 @@ def self.add_destroy_callbacks(model, reflection)
       name = reflection.name
       model.after_destroy lambda { |o| o.association(name).handle_dependency }
     end
+
+    def self.define_validations(model, reflection)
+      if reflection.options.key?(:required)
+        ActiveSupport::Deprecation.warn('`required` option for belongs_to relation is deprecated and replaced with `optional` option.')
```

Comment:
That was the old default, which means no client code will have that code, since it doesn’t make any sense.

> On Feb 15, 2015, at 12:54 PM, Josef Šimánek <notifications@github.com> wrote:
> 
> In activerecord/lib/active_record/associations/builder/belongs_to.rb <https://github.com/rails/rails/pull/18937#discussion_r24725940>:
> 
> > @@ -110,5 +110,24 @@ def self.add_destroy_callbacks(model, reflection)
> >        name = reflection.name
> >        model.after_destroy lambda { |o| o.association(name).handle_dependency }
> >      end
> > +
> > +    def self.define_validations(model, reflection)
> > +      if reflection.options.key?(:required)
> > +        ActiveSupport::Deprecation.warn('`required` option for belongs_to relation is deprecated and replaced with `optional` option.')
> What about when required: false is specified explicitly? Maybe throw deprecation warning only for false value?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/rails/rails/pull/18937/files#r24725940>.
>